packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -0,0 +1,41 @@
+package expo.modules.image
+
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.model.GlideUrl
+import com.bumptech.glide.request.FutureTarget
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runInterruptible
+import java.lang.Exception
+
+suspend fun <T> FutureTarget<T>.awaitGet() = runInterruptible(Dispatchers.IO) { get() }
+
+class ExpoImageModule(val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+  private val moduleCoroutineScope = CoroutineScope(Dispatchers.IO)
+  override fun getName() = "ExpoImageModule"
+
+  @ReactMethod
+  fun prefetch(url: String, promise: Promise) {
+    moduleCoroutineScope.launch {
+      try {
+        val glideUrl = GlideUrl(url)
+        val result = Glide.with(context)
+            .download(glideUrl)
+            .submit()
+            .awaitGet()
+        if (result != null) {
+          promise.resolve(null)
+        } else {
+          promise.reject("ERR_IMAGE_PREFETCH_FAILURE", "Failed to prefetch the image: ${url}.")
+        }
+      } catch (e: Exception) {
+        promise.reject("ERR_IMAGE_PREFETCH_FAILURE", "Failed to prefetch the image: ${e.message}", e)
+      }
+    }
+  }
```

Comment:
Have you tried doing it this way? I think we should create the `Future` in the "main" thread, but only wait for it in coroutine. Additionally, putting `coroutineScope.launch { }` inside `try-catch` will catch possible `CancellationException` which will be useful when you'll be adding `abortPrefetch` method.

```suggestion
  fun prefetch(url: String, promise: Promise) {
    try {
      val glideUrl = GlideUrl(url)
      val future = Glide.with(context)
          .download(glideUrl)
          .submit()
            
      moduleCoroutineScope.launch {    
        val result = future.awaitGet()
        if (result != null) {
          promise.resolve(null)
        } else {
          promise.reject("ERR_IMAGE_PREFETCH_FAILURE", "Failed to prefetch the image: ${url}.")
        }
      }
    } catch (e: Exception) {
      promise.reject("ERR_IMAGE_PREFETCH_FAILURE", "Failed to prefetch the image: ${e.message}", e)
    }
  }
```